### PR TITLE
Add a golden scenario test for fallback font rendering on iOS take 2

### PR DESCRIPTION
**Pull request by xster**
Didn't have time to revise https://github.com/flutter/engine/pull/20687 and merge it before. This would have actually helped catch https://github.com/flutter/flutter/issues/68399 too I believe. 

Original PR relates to flutter/flutter#60013. If user specifies a font that doesn't exist, it should properly fallback to the SF font and render it correctly.

Also removing the duplicate iPhone SE images as discussed with @blasten and refactor a modular golden test part out of the platform view stuff. 

Also addresses https://github.com/flutter/flutter/issues/68625.

### CODE REVIEW

**Comment by xster** [comment]
I think the previous pre-submit failures were due to our LUCI bots re-running the golden generation on iOS 13 whereas I made it on iOS 14. Might have to download iOS 13 again.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m [hunk 1]
```diff
@@ -44,4 +44,29 @@ - (instancetype)initWithLaunchArg:(NSString*)launchArg {
...
+    attachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+    [test addAttachment:attachment];
+    XCTFail(@"This test will fail - no golden named %@ found. Follow the steps in the "
+            @"README to add a new golden.",
+            _goldenImage.goldenName);
+  }
+
+  if (![_goldenImage compareGoldenToImage:screenshot.image]) {
+    XCTAttachment* screenshotAttachment;
+    screenshotAttachment = [XCTAttachment attachmentWithImage:screenshot.image];
```
Comment (blasten) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m:61]: nit: these two can be in a single line

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m [hunk 2]
```diff
@@ -44,4 +44,29 @@ - (instancetype)initWithLaunchArg:(NSString*)launchArg {
...
+    screenshotAttachment.name = _goldenImage.goldenName;
+    screenshotAttachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+    [test addAttachment:screenshotAttachment];
+
+    XCTFail(@"Goldens to not match. Follow the steps in the "
```
Comment (blasten) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m:66]: nit: do not match?

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m [hunk 3]
```diff
@@ -44,4 +44,29 @@ - (instancetype)initWithLaunchArg:(NSString*)launchArg {
...
+    [test addAttachment:attachment];
+    XCTFail(@"This test will fail - no golden named %@ found. Follow the steps in the "
+            @"README to add a new golden.",
+            _goldenImage.goldenName);
+  }
+
+  if (![_goldenImage compareGoldenToImage:screenshot.image]) {
+    XCTAttachment* screenshotAttachment;
+    screenshotAttachment = [XCTAttachment attachmentWithImage:screenshot.image];
+    screenshotAttachment.name = _goldenImage.goldenName;
```
Comment (blasten) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m:62]: maybe format this as make it `_goldenImage.goldenName + "_current"`

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m [hunk 4]
```diff
@@ -44,4 +44,29 @@ - (instancetype)initWithLaunchArg:(NSString*)launchArg {
   return self;
 }
 
+- (void)checkGoldenForTest:(XCTestCase*)test {
+  XCUIScreenshot* screenshot = [[XCUIScreen mainScreen] screenshot];
+  if (!_goldenImage.image) {
+    XCTAttachment* attachment = [XCTAttachment attachmentWithScreenshot:screenshot];
+    attachment.name = @"new_golden";
```
Comment (blasten) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m:51]: similar: maybe format this as make it `_goldenImage.goldenName + "_new"`

File: testing/scenario_app/ios/Scenarios/Scenarios.xcodeproj/xcshareddata/xcschemes/Scenarios.xcscheme [hunk 5]
```diff
@@ -28,15 +28,6 @@
...
       systemAttachmentLifetime = "keepNever">
-      <MacroExpansion>
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "248D76C622E388370012F0C1"
-            BuildableName = "Scenarios.app"
-            BlueprintName = "Scenarios"
-            ReferencedContainer = "container:Scenarios.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
```
Comment (blasten) [inline testing/scenario_app/ios/Scenarios/Scenarios.xcodeproj/xcshareddata/xcschemes/Scenarios.xcscheme:39]: what does this do?

**Review by blasten — APPROVED** [review]
LGTM + nits

**Review by xster — COMMENTED** [review]
I have no idea how but I weaksauced the last PR because the golden screenshot I saved was in the wrong dpi.

**Comment by xster** [comment]
oops, forgot the tree is closed

**Comment by xster** [comment]
Tests are crapping out probably due to some transient nnbd issue. Rebasing

**Comment by chinmaygarde** [comment]
There are conflicts to the auto-merge bot won't land this. @xster can you please rebase? The tree is open now.

**Pull request merged**